apps/src/lib/script-editor/FlexGroup.jsx

```diff
@@ -0,0 +1,219 @@
+    padding: 10,
+    marginBottom: 20
+  },
+  addGroup: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none',
+    margin: '0 0 30px 0'
+  },
+  addStage: {
+    fontSize: 14,
+    color: '#5b6770',
+    background: 'white',
+    border: '1px solid #ccc',
+    boxShadow: 'none',
+    margin: '0 10px 10px 10px'
+  }
+};
+
+class FlexGroup extends Component {
+  static propTypes = {
+    addGroup: PropTypes.func.isRequired,
+    addStage: PropTypes.func.isRequired,
+    stages: PropTypes.array.isRequired,
+    levelKeyList: PropTypes.object.isRequired
+  };
+
+  handleAddGroup = () => {
+    this.props.addGroup(
+      prompt('Enter new stage name'),
+      prompt('Enter new group name')
+    );
+  };
+
+  handleAddStage = position => {
+    this.props.addStage(position, prompt('Enter new stage name'));
+  };
+
+  /**
+   * Generate the ScriptDSL format.
+   * @param stages
+   * @return {string}
+   */
+  serializeStages = stages => {
+    let s = [];
+    stages.forEach(stage => {
+      let t = `stage '${stage.name}'`;
+      if (stage.lockable) {
+        t += ', lockable: true';
+      }
+      if (stage.flex_category) {
+        t += `, flex_category: '${stage.flex_category}'`;
+      }
+      s.push(t);
+      stage.levels.forEach(level => {
+        if (level.ids.length > 1) {
+          s.push('variants');
+          level.ids.forEach(id => {
+            s = s.concat(this.serializeLevel(id, level, level.activeId === id));
+          });
+          s.push('endvariants');
+        } else {
+          s = s.concat(this.serializeLevel(level.ids[0], level));
+        }
+      });
+      s.push('');
+    });
+    return s.join('\n');
+  };
+
+  serializeLevel(id, level, active) {
+    if (id === NEW_LEVEL_ID) {
+      return;
+    }
+
+    const s = [];
+    const key = this.props.levelKeyList[id];
+    if (/^blockly:/.test(key)) {
+      if (level.skin) {
+        s.push(`skin '${level.skin}'`);
+      }
+      if (level.videoKey) {
+        s.push(`video_key_for_next_level '${level.videoKey}'`);
+      }
+      if (level.concepts) {
+        s.push(`concepts ${level.concepts}`);
+      }
+      if (level.conceptDifficulty) {
+        s.push(`level_concept_difficulty '${level.conceptDifficulty}'`);
+      }
+    }
+    let l = `${this.normalizeLevelKind(level.kind)} '${key.replace(
+      /'/,
+      "\\'"
+    )}'`;
+    if (active === false) {
+      l += ', active: false';
+    }
+    if (level.progression) {
+      l += `, progression: '${level.progression}'`;
+    }
+    s.push(l);
+    return s;
+  }
+
+  /**
+   * Levels with kind "puzzle" and "unplugged" are special cases of "level", for
+   * the purpose of the ScriptDSL.
+   * @param kind
+   * @return {string}
+   */
+  normalizeLevelKind(kind) {
+    return !kind || kind === LevelKind.puzzle || kind === LevelKind.unplugged
+      ? LevelKind.level
+      : kind;
```

Comment:
I'm not sure LevelKind is the right thing to group levels by here -- for example, I don't think a levelbuilder needs to care if they are picking a puzzle vs an unplugged level since these are treated the same. am I missing something here?

related: we want to be able to set multiple script level properties, e.g. "named" and "assessment", so I'll need to change the UX to move away from a single menu.